pkg/resource/v1/chartconfig/current.go

```diff
@@ -2,8 +2,45 @@ package chartconfig
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GetCurrentState returns the ChartConfig resources present in the guest
+// cluster.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for chartconfigs in the guest cluster")
+
+	clusterConfig, err := prepareClusterConfig(r.baseClusterConfig, clusterGuestConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	g8sClient, err := r.guestClusterService.GetG8sClient(ctx, clusterConfig.ClusterID, clusterConfig.Domain.API)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	chartConfigList, err := g8sClient.CoreV1alpha1().ChartConfigs(metav1.NamespaceSystem).List(metav1.ListOptions{})
```

Comment:
I'm checking for the chartconfigs in the `kube-system` namespace. We've already agreed chart-operator will run in kube-system on the master node. This is to isolate it from customers and their workloads. I think the same applies to the CRs.